### PR TITLE
Nessie: Avoid usage of deprecated APIs in test

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Deprecated API usage:

image

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
cc: @dimas-b

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java [hunk 1]
```diff
@@ -114,8 +113,8 @@ private void resetData() throws NessieConflictException, NessieNotFoundException
   }
 
   @BeforeEach
-  public void beforeEach(@NessieUri URI nessieUri) throws IOException {
-    this.uri = nessieUri.toString();
+  public void beforeEach(@NessieClientUri URI nessieUri) throws IOException {
+    this.uri = nessieUri.resolve("v1").toASCIIString();
```
Comment (dimas-b) [inline nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java:117]: With `@NessieClientUri` it is preferable to also use `@NessieApiVersions` (cf. examples in the Nessie repo). This way, the test framework will inject the correct URI for the API version.

**Review by dimas-b — COMMENTED** [review]
Thanks for looking into this @ajantha-bhat !

Please consider moving to multi-version tests. The old annotations are deprecated specifically to indicate the need to use the multi-version test extension.

I think initially it is fine to limit the PR to Nessie API v1. When `NessieCatalog` supports Nessie API v2 it will be easy to upgrade tests to cover both use cases.

File: nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java (hunk 1 above)
Comment (ajantha-bhat) [reply nessie/src/test/java/org/apache/iceberg/nessie/BaseTestIceberg.java]: Thanks for the review. I have updated the code to use `NessieApiVersions` now

**Comment by ajantha-bhat** [comment]
> Please consider moving to multi-version tests. The old annotations are deprecated specifically to indicate the need to use the multi-version test extension.
I think initially it is fine to limit the PR to Nessie API v1. When NessieCatalog supports Nessie API v2 it will be easy to upgrade tests to cover both use cases.

Thanks for the suggestion. Moved to multi-version tests now.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java [hunk 2]
```diff
@@ -70,12 +72,14 @@ public class TestNessieCatalog extends CatalogTests<NessieCatalog> {
...
-    this.uri = nessieUri.toString();
-    this.api = HttpClientBuilder.builder().withUri(this.uri).build(NessieApiV1.class);
-
-    initialHashOfDefaultBranch = api.getDefaultBranch().getHash();
-
+  public void setUp(NessieClientFactory clientFactory, @NessieClientUri URI nessieUri) {
+    api = clientFactory.make();
+    try {
+      initialHashOfDefaultBranch = api.getDefaultBranch().getHash();
+    } catch (NessieNotFoundException e) {
```
Comment (dimas-b) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java:79]: Why not declare the exception as thrown by this method? JUnit 5 can deal with that and the code will be simpler and easier to read :thinking:

**Review by dimas-b — COMMENTED** [review]
Test changes LGTM :+1: Just one minor style / readability comment.

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java (hunk 2 above)
Comment (ajantha-bhat) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java]: done

**Comment by ajantha-bhat** [comment]
@dimas-b: Thanks for the review. @Fokko: Can you please help in reviewing and merging the PR?

**Review by dimas-b — APPROVED** [review]
:+1:

**Review by Fokko — APPROVED** [review]
Thanks for picking this up @ajantha-bhat, much appreciated! And thanks @dimas-b for the review! Going in!

**Pull request merged**